```
error: failed to run custom build command for `gecko-media v0.1.0 (https://github.com/servo/gecko-media.git#718662a0)`
process didn't exit successfully: `/home/servo/buildbot/slave/linux-rel-wpt/build/target/release/build/gecko-media-0b83353c9864ffb9/build-script-build` (exit code: 101)
--- stdout
running: "cmake" "/home/servo/.cargo/git/checkouts/gecko-media-6bedb392a81eb7cf/718662a/." "-DCMAKE_INSTALL_PREFIX=/home/servo/buildbot/slave/linux-rel-wpt/build/target/release/build/gecko-media-d2681dd88a6f4481/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64" "-DCMAKE_CXX_COMPILER=/usr/bin/c++" "-DCMAKE_BUILD_TYPE=Release"
-- Configuring incomplete, errors occurred!

--- stderr
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.1 or higher is required.  You are running version 2.8.12.2


thread 'main' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', /home/servo/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.26/src/lib.rs:599:4
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:69
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:58
             at /checkout/src/libstd/panicking.rs:381
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:397
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:577
   5: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:538
   6: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:522
   7: cmake::fail
   8: cmake::run
   9: cmake::Config::build
  10: build_script_build::main
  11: __rust_maybe_catch_panic
             at /checkout/src/libpanic_unwind/lib.rs:99
  12: std::rt::lang_start
             at /checkout/src/libstd/panicking.rs:459
             at /checkout/src/libstd/panic.rs:361
             at /checkout/src/libstd/rt.rs:59
  13: __libc_start_main
  14: <unknown>

warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,6 +18,7 @@ matrix:
         - sudo apt-get update -q
         - sudo apt-get install clang-3.9 llvm-3.9 llvm-3.9-runtime -y
         - export LLVM_CONFIG=/usr/lib/llvm-3.9/bin/llvm-config
+        - export CC=gcc-5 CXX=g++-5
       script:
          - ./mach build -d --verbose
          - ./mach test-compiletest
@@ -37,12 +38,24 @@ matrix:
       env: CCACHE=/usr/bin/ccache
       addons:
         apt:
+          sources:
+            - ubuntu-toolchain-r-test
+            - sourceline: 'ppa:jonathonf/ffmpeg-3'
           packages:
             - cmake
+            - dbus-x11
             - freeglut3-dev
+            - gcc-5
+            - g++-5
             - gperf
+            - libavcodec-dev
+            - libavformat-dev
             - libosmesa6-dev
+            - libpulse-dev
             - libgles2-mesa-dev
+            - libswscale-dev
+            - libswresample-dev
+            - pulseaudio
             - python-virtualenv
             - xorg-dev
             - ccache

--- a/components/script/Cargo.toml
+++ b/components/script/Cargo.toml
@@ -23,6 +23,9 @@ phf_codegen = "0.7.18"
 phf_shared = "0.7.18"
 serde_json = "1.0"
 
+[target.'cfg(all(any(target_os = "macos", target_os = "linux"), not(any(target_arch = "arm", target_arch = "aarch64"))))'.dependencies]
+gecko-media = {git = "https://github.com/servo/gecko-media.git"}
+
 [target.'cfg(any(target_os = "macos", target_os = "linux", target_os = "windows"))'.dependencies]
 tinyfiledialogs = "2.5.9"
 

--- a/components/script/lib.rs
+++ b/components/script/lib.rs
@@ -41,6 +41,9 @@ extern crate domobject_derive;
 extern crate encoding_rs;
 extern crate euclid;
 extern crate fnv;
+#[allow(unused_extern_crates)]
+#[cfg(all(any(target_os = "macos", target_os = "linux"), not(any(target_arch = "arm", target_arch = "aarch64"))))]
+extern crate gecko_media;
 extern crate gleam;
 extern crate half;
 #[macro_use] extern crate html5ever;

--- a/etc/ci/buildbot_steps.yml
+++ b/etc/ci/buildbot_steps.yml
@@ -61,12 +61,12 @@ mac-nightly:
 
 linux-rel-intermittent:
   - ./mach clean-nightlies --keep 3 --force
-  - ./mach build --release
+  - env CC=gcc-5 CXX=g++-5 ./mach build --release
   - ./etc/ci/check_intermittents.sh --log-raw intermittents.log
 
 linux-rel-nogate:
   - ./mach clean-nightlies --keep 3 --force
-  - ./mach build --release
+  - env CC=gcc-5 CXX=g++-5 ./mach build --release
   - python ./etc/ci/chaos_monkey_test.py
 
 mac-rel-intermittent:
@@ -78,7 +78,7 @@ linux-dev:
   - ./mach clean-nightlies --keep 3 --force
   - ./mach test-tidy --no-progress --all
   - ./mach test-tidy --no-progress --self-test
-  - env ./mach build --dev
+  - env CC=gcc-5 CXX=g++-5 ./mach build --dev
   - env ./mach test-compiletest
   - env ./mach test-unit
   - env ./mach package --dev
@@ -92,7 +92,7 @@ linux-dev:
 
 linux-rel-wpt:
   - ./mach clean-nightlies --keep 3 --force
-  - ./mach build --release --with-debug-assertions
+  - env CC=gcc-5 CXX=g++-5 ./mach build --release --with-debug-assertions
   - ./mach test-wpt-failure
   - ./mach test-wpt --release --processes 24 --total-chunks 2 --this-chunk 1 --log-raw test-wpt.log --log-errorsummary wpt-errorsummary.log --always-succeed
   - ./mach filter-intermittents wpt-errorsummary.log --log-intermittents intermittents.log --log-filteredsummary filtered-wpt-errorsummary.log --tracker-api default --reporter-api default
@@ -100,7 +100,7 @@ linux-rel-wpt:
 
 linux-rel-css:
   - ./mach clean-nightlies --keep 3 --force
-  - ./mach build --release --with-debug-assertions
+  - env CC=gcc-5 CXX=g++-5 ./mach build --release --with-debug-assertions
   - ./mach test-wpt --release --processes 24 --total-chunks 2 --this-chunk 2 --log-raw test-wpt.log --log-errorsummary wpt-errorsummary.log --always-succeed
   - ./mach filter-intermittents wpt-errorsummary.log --log-intermittents intermittents.log --log-filteredsummary filtered-wpt-errorsummary.log --tracker-api default --reporter-api default
   - ./mach build-geckolib --release
@@ -110,7 +110,7 @@ linux-rel-css:
 
 linux-nightly:
   - ./mach clean-nightlies --keep 3 --force
-  - ./mach build --release
+  - env CC=gcc-5 CXX=g++-5 ./mach build --release
   - ./mach package --release
   - ./mach upload-nightly linux
 

--- a/servo-tidy.toml
+++ b/servo-tidy.toml
@@ -35,6 +35,10 @@ num = []
 # Ignored packages with duplicated versions
 packages = [
   "bitflags",
+  "mime",
+  "semver",
+  "toml",
+  "unicase",
 ]
 # Files that are ignored for all tidy and lint checks.
 files = [

